### DISCUSSION TITLE
Types / `rbs` / `sorbet` / `typeprof`

### DISCUSSION THREAD

**Discussion by marcandre** [rubocop/rubocop#9214] [Ideas]
In many cases, Cops make decisions on the name of a method and assume the class of the receiver. 

We should start thinking of pluging into `rbs` and being able to ask `hey, this receiver here, can you tell me what's the type?`. If it is `untyped`, then we proceed as we currently do, but otherwise we could either be sure it's an offense, or sure [this is not the method we were expecting](https://github.com/rubocop-hq/rubocop/pull/9112#issuecomment-742986717).

**Comment by marcandre** [comment]
I haven't looked into `rbs` enough yet, I believe it has information only on method signatures, but `typeprof` is aimed to deduce the type of all the inner variables, so maybe there's a way to extract the information? Seems difficult as I don't think there's an export format for that type of information, and `typeprof` does not use `parser` so it would definitely be a huge challenge. Maybe for 2030 😅

**Comment by MGatner** [comment]
Does Rubocop have any Cops that enforce RBS signatures? I looked around and landed here.
